src/journal/cat.c

```diff
@@ -151,8 +155,23 @@ static int run(int argc, char *argv[]) {
 
         if (argc <= optind)
                 (void) execl("/bin/cat", "/bin/cat", NULL);
-        else
+        else {
+                _cleanup_free_ char *s = NULL;
+                struct stat st;
+
+                if (fstat(has_errfd ? STDERR_FILENO : STDOUT_FILENO, &st) < 0)
```

Comment:
hmm, the ternary op is unnecessary afaics. rearrange_stdio() placed the stream fd to stderr anyway, hence you can just unconditionally use STDERR_FILENO here.